### PR TITLE
Add where clause to filter out test user email from devlocal service member users list

### PR DESCRIPTION
**Pull request by duncan-truss**
## Summary

Before A-team starts outcome work on PPM, I thought it would be good to filter out all of the test users from the devlocal sign in screen that are created when populating the database with the e2e devseed.  These are users that have the default email `first.last@login.gov.test` when the devseed calls MakeUser/MakeServiceMember without any override assertions.  When we do create PPM specific data usually the email describes the purpose such as `needs@orde.rs`.  The devlocal login sorts users by last created date and there is a user limit so it often hides the most useful premade users.

The fetchAppUserIdentities method is only used for querying the users for devlocal so does not affect any real app usage.

This doesn't impact the office app devlocal sign in list only the customer app. 

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Populate the develepment database with the e2e devseed scenario

```sh
make db_dev_e2e_populate
```

##### Terminal 2

Start the UI locally.

```sh
make client_run
```

##### Terminal 3

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

1. In your browser open http://milmovelocal:3000
2. Select Local Sign In from the homepage 
3. The users list should only contain the custom service member users to select from and not those created by the office.

## Verification Steps for Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Request review from a member of a different team.
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide/#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#zero-downtime-migrations)
- [ ] Have been communicated to #g-database
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#secure-migrations)

## Screenshots

Before

Screen Shot 2022-01-14 at 10 40 56 AM

After

Screen Shot 2022-01-14 at 10 36 00 AM

### CODE REVIEW

File: pkg/models/user_test.go [hunk 1]
```diff
@@ -257,29 +257,32 @@ func (suite *ModelSuite) TestFetchAppUserIdentities() {
...
-		suite.NotEmpty(identities)
-		suite.Equal(1, len(identities))
 
-		if len(identities) > 1 {
-			suite.NotNil(identities[0].ServiceMemberID)
-			suite.Nil(identities[0].OfficeUserID)
-		}
+		identities, err := FetchAppUserIdentities(suite.DB(), auth.MilApp, 5)
 
-		// Service member is super user
```
Comment (duncan-truss) [inline pkg/models/user_test.go:272]: There used to be a column called is_superuser on the users table but it was removed a while ago and this test wasn't doing anything so I'm deleting it.

**Comment by ronaktruss** [comment]
@duncan-truss Are the user with first.last@login.gov.test not used for other parts of the application / are there other more specific logins that serve the same purpose? I know I had used some of these users when working on load testing.

**Review by felipe-lee — APPROVED** [review]
I like that this is a quick solution to this for the purpose of seeing useful customers to log in as, but I also think we still have an underlying issue that most of these users are also mistakenly created. People don't always realize that several testdatagen function will create a new user if you don't pass it in and will call several of these in a row which creates several users when they only needed one user.

**Comment by duncan-truss** [comment]
> @duncan-truss Are the user with [first.last@login.gov.test](mailto:first.last@login.gov.test) not used for other parts of the application / are there other more specific logins that serve the same purpose? I know I had used some of these users when working on load testing.

@ronaktruss This user email is only from data created in the devseed (make db_dev_e2e_populate) and e2e_basic (make db_e2e_up) so it adds data for local/ephemeral deploys/and in Cypress tests.  It's the byproduct of creating moves used in the office for the most part where we never intend to login as the service member but there must be some record for the move to exist.

In load testing I think you are calling [createUser](https://github.com/transcom/mymove/blob/6ed72bf51a630cf834d188b707fc46f302079084/pkg/auth/authentication/devlocal.go#L410) which creates a unique email (timestamp+nonce@example.com) based on the timestamp if you aren't overriding it, same as clicking the "Create a new Milmove User" button.

You could still enter the email into the login with user email if you had one unique user with that email but it becomes useless when you have multiple with the same email.  We should probably add an input to login by user id, which is how the user list works.

**Comment by duncan-truss** [comment]
> I like that this is a quick solution to this for the purpose of seeing useful customers to log in as, but I also think we still have an underlying issue that most of these users are also mistakenly created. People don't always realize that several testdatagen function will create a new user if you don't pass it in and will call several of these in a row which creates several users when they only needed one user.

@felipe-lee That's a good point I wasn't thinking so much about the test factories that say create multiple users incorrectly.  I am down to look at that further as we work on PPMs.

**Pull request merged**